### PR TITLE
[stable/nginx] adding serviceAccount to controller for better rbac support

### PR DESCRIPTION
**Pull request by thomas-b-jackson**
We would like to start using this chart with the 0.9 version of the nginx-ingress which allows namespace-specific controllers. In order for that to be useful, we need to run the nginx deployment under a namespace-specific serviceaccount that cluster administrators have bound to a "least privileges" cluster role, as defined [here](https://github.com/kubernetes/ingress/blob/master/examples/rbac/nginx/nginx-ingress-controller-rbac.yml#L12).

### CODE REVIEW

**Comment by thomas-b-jackson** [comment]
I don't understand the "commit missing GitHub user" error. I have signed into the cla/linuxfoundation as my thomas.b.jackson@gmail.com user.

**Comment by unguiculus** [comment]
Author and committer e-mail are different (your Nordstrom address).

File: stable/nginx-ingress/templates/controller-daemonset.yaml
Comment (prydonius) [inline stable/nginx-ingress/templates/controller-daemonset.yaml]: use `serviceAccountName` instead, `serviceAccount` is deprecated https://kubernetes.io/docs/api-reference/v1.6/#podspec-v1-core

**Review by prydonius — DISMISSED** [review]
awesome thanks! can you also bump the chart minor version?

File: stable/nginx-ingress/README.md
Comment (mgoodness) [inline stable/nginx-ingress/README.md]: Since (as @prydonius noted) the field's name is `serviceAccountName`, let's use that as the key name.

File: stable/nginx-ingress/templates/controller-daemonset.yaml
Comment (mgoodness) [reply stable/nginx-ingress/templates/controller-daemonset.yaml]: Since we don't actually _need_ a value, let's just do:
```
{{- if .Values.controller.serviceAccountName }}
  serviceAccountName: {{ .Values.controller.serviceAccountName }}
{{- end }}
```

File: stable/nginx-ingress/templates/controller-deployment.yaml
Comment (mgoodness) [inline stable/nginx-ingress/templates/controller-deployment.yaml]: Refer to DaemonSet comment.

**Review by mgoodness — CHANGES_REQUESTED** [review]

**Comment by thomas-b-jackson** [comment]
Hey peeps, 
Thanks for the great suggestions. Mo better now?

**Comment by unguiculus** [comment]
Please rebase. Otherwise LGTM.

**Review by mgoodness — APPROVED** [review]

**Comment by mgoodness** [comment]
LGTM 2 after rebase.

**Pull request merged**
